File: sub-web/src/main/java/org/github/controller/MenuController.java
package org.github.controller;

import org.github.mysql.web.base.entity.SysMenuEntity;
import org.github.spring.restful.Returnable;
import org.github.spring.restful.json.JSONPageReturn;
import org.github.web.module.sys.menu.model.bo.QueryMenuListBO;
import org.github.web.module.sys.menu.service.IMenuService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RequestMapping("/sys/menu/")
@Controller
public class MenuController {
  @Autowired
  private IMenuService menuService;

  @PostMapping("del")
  public Returnable delMenuList(@RequestBody List<Long> ids) {
    return menuService.delMenuList(ids);
  }

  @GetMapping("info/{menuId}")
  public Returnable queryMenuInfo(@PathVariable Long menuId) {
    return menuService.queryMenuInfo(menuId);
  }

  @PostMapping("list")
  public Returnable queryMenuList(@RequestBody QueryMenuListBO bo) {
    return JSONPageReturn.of(menuService.queryMenuList(bo));
  }

  @PostMapping("select")
  public Returnable queryMenuTree() {
    return menuService.queryMenuTree();
  }

  @PostMapping({"save", "update"})
  public Returnable saveOrUpdate(@RequestBody SysMenuEntity bo) {
    return menuService.saveOrUpdate(bo);
  }
}

File: sub-web/src/main/java/org/github/controller/MenuController.kt
package org.github.controller

import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.springframework.beans.factory.annotation.Autowired
import org.github.web.module.sys.menu.service.IMenuService
import org.github.spring.restful.Returnable
import org.github.web.module.sys.menu.model.bo.QueryMenuListBO
import org.github.mysql.mydb.base.entity.SysMenuEntity
import org.github.ops.trim
import org.springframework.web.bind.annotation.*
import org.springframework.web.bind.annotation.RestController

@Api(tags = ["菜单管理"])
@RequestMapping("/sys/menu/")
@RestController
class MenuController {
  @Autowired
  private lateinit var menuService: IMenuService

  @ApiOperation("删除菜单")
  @PostMapping("del")
  fun delMenuList(@RequestBody ids: Array<Long>): Returnable {
    return menuService.delMenuList(ids.toList())
  }

  @GetMapping("info/{menuId}")
  fun queryMenuInfo(@PathVariable menuId: Long): Returnable {
    return menuService.queryMenuInfo(menuId)
  }

  @ApiOperation("分页查询")
  @PostMapping("page")
  fun queryMenuPage(@RequestBody bo: QueryMenuListBO): Returnable {
    return menuService.queryMenuPage(bo.trim())
  }

  @PostMapping("select")
  fun queryMenuTree(): Returnable {
    return menuService.queryMenuTree()
  }

  @ApiOperation("新增/修改")
  @PostMapping("save", "update")
  fun saveOrUpdate(@RequestBody bo: SysMenuEntity): Returnable {
    return menuService.saveOrUpdate(bo.trim())
  }
}
